recipes-debian/AppArmor/apparmor_debian.bb

```diff
@@ -0,0 +1,204 @@
+    #| Makefile:478: recipe for target 'Makefile.perl' failed
+    sed -i "s@sed -ie 's///g' Makefile.perl@@" ${S}/libraries/libapparmor/swig/perl/Makefile
+
+
+    oe_runmake -C ${B}/libraries/libapparmor
+    oe_runmake -C ${B}/binutils
+    oe_runmake -C ${B}/utils
+    oe_runmake -C ${B}/parser
+    oe_runmake -C ${B}/profiles
+
+    if test -z "${HTTPD}" ; then
+    oe_runmake -C ${B}/changehat/mod_apparmor
+    fi
+
+    if test -z "${PAMLIB}" ; then
+    oe_runmake -C ${B}/changehat/pam_apparmor
+    fi
+}
+
+do_install () {
+    install -d ${D}/${INIT_D_DIR}
+    install -d ${D}/lib/apparmor
+
+    oe_runmake -C ${B}/libraries/libapparmor DESTDIR="${D}" install
+    oe_runmake -C ${B}/binutils DESTDIR="${D}" install
+    oe_runmake -C ${B}/utils DESTDIR="${D}" install
+    oe_runmake -C ${B}/parser DESTDIR="${D}" install
+    oe_runmake -C ${B}/profiles DESTDIR="${D}" install
+
+    # If perl is disabled this script won't be any good
+    if ! ${@bb.utils.contains('PACKAGECONFIG','perl','true','false', d)}; then
+    rm -f ${D}${sbindir}/aa-notify
+    fi
+
+    if test -z "${HTTPD}" ; then
+    oe_runmake -C ${B}/changehat/mod_apparmor DESTDIR="${D}" install
+    fi
+
+    if test -z "${PAMLIB}" ; then
+    oe_runmake -C ${B}/changehat/pam_apparmor DESTDIR="${D}" install
+    fi
+
+    # aa-easyprof is installed by python-tools-setup.py, fix it up
+    sed -i -e 's:/usr/bin/env.*:/usr/bin/python3:' ${D}${bindir}/aa-easyprof
+    chmod 0755 ${D}${bindir}/aa-easyprof
+
+    install ${WORKDIR}/apparmor ${D}/${INIT_D_DIR}/apparmor
+    install ${WORKDIR}/functions ${D}/lib/apparmor
+    install -d ${D}${systemd_system_unitdir}
+    install ${WORKDIR}/apparmor.service ${D}${systemd_system_unitdir}
+}
+
+do_install_append() {
+    #Install files add by debian package
+    #debian/apparmor.install
+    install -d ${D}/usr/share/apport/package-hooks
+    install ${S}/debian/apport/source_apparmor.py ${D}/usr/share/apport/package-hooks
+    install -d ${D}/usr/share/apparmor-features
+    install ${S}/debian/features ${D}/usr/share/apparmor-features
+    install ${S}/debian/lib/apparmor/profile-load ${D}/lib/apparmor
+    install ${S}/parser/apparmor.systemd ${D}/lib/apparmor
+    install ${S}/parser/rc.apparmor.functions ${D}/lib/apparmor/
+    install ${S}/parser/aa-teardown ${D}${sbindir}
+    install -D ${S}/debian/apparmor.lintian-overrides ${S}/usr/share/lintian/overrides/apparmor
+    #debian/apparmor-utils.install
+    install ${S}/debian/aa-update-browser ${D}${sbindir}
+    install -D ${S}/debian/vim-apparmor.yaml ${D}/usr/share/vim/registry
+    install -D ${D}/usr/share/apparmor/apparmor.vim ${D}/usr/share/vim/addons/syntax
+    install -D ${S}/debian/apparmor-utils.lintian-overrides ${S}/usr/share/lintian/overrides/apparmor-utils
+    #debian/apparmor-easyprof.install
+    install -D ${S}/debian/apparmor-easyprof.lintian-overrides ${S}/usr/share/lintian/overrides/apparmor-easyprof
+}
+
+#Building ptest on arm fails.
```

Comment:
Please describe more information (e.g. "what kind of error you got", "is there any way to fix the error or not",  and so on) in this recipe or commit log.

Reply:
[meta-security's recipe](https://git.yoctoproject.org/meta-security/tree/recipes-mac/AppArmor/apparmor_2.13.2.bb?h=warrior) doesn't have such functions. Does meta-security has same problem?

Reply:
meta-security has the same problem.
Backported the fix commit.
https://git.yoctoproject.org/meta-security/commit/recipes-mac/AppArmor?id=27ddb455543b670097e252ba0d0ad5b7e4101748